policy-engine/src/main.rs

```diff
@@ -72,33 +75,59 @@ async fn main() -> Result<(), Error> {
     let registry: &'static Registry = Box::leak(Box::new(metrics::new_registry(Some(
         METRICS_PREFIX.to_string(),
     ))?));
-    graph::register_metrics(registry)?;
     registry.register(Box::new(BUILD_INFO.clone()))?;
+
+    // Main service.
+    let plugins = settings.validate_and_build_plugins(Some(registry))?;
+
+    // Shared state.
+    let state = {
+        let mandatory_params = settings.mandatory_client_parameters.clone();
+        let path_prefix = settings.path_prefix.clone();
+        let plugins = Box::leak(Box::new(plugins));
+        let live = Arc::new(RwLock::new(false));
+        let ready = Arc::new(RwLock::new(false));
+
+        AppState::new(
+            mandatory_params,
+            path_prefix,
+            plugins,
+            live,
+            ready,
+            registry,
+        )
+    };
+
+    graph::register_metrics(state.registry())?;
+    let metric_state = state.clone();
     let metrics_server = HttpServer::new(move || {
         App::new()
             .wrap(middleware::Compress::default())
-            .app_data(actix_web::web::Data::new(RegistryWrapper(registry)))
+            .app_data(actix_web::web::Data::new(metric_state.clone()))
             .service(
                 actix_web::web::resource("/metrics")
-                    .route(actix_web::web::get().to(metrics::serve::<RegistryWrapper>)),
+                    .route(actix_web::web::get().to(metrics::serve::<AppState>)),
+            )
+            .service(
+                actix_web::web::resource("/liveness")
+                    .route(actix_web::web::get().to(status::serve_liveness)),
+            )
+            .service(
+                actix_web::web::resource("/readiness")
+                    .route(actix_web::web::get().to(status::serve_readiness)),
             )
     })
     .bind((settings.status_address, settings.status_port))?
     .run();
 
+    // metrics endpoints has started running
+    *state.live.write() = true;
```

Comment:
This seems like a brittle way to say we're alive now, because it won't change if something happens in the container that would benefit from restarting the container.  But I'm not sure when the container would benefit from a restart.  Possibly if the graph-data cache gets stale, a fresh container might have better luck connecting?  But if the issue is on the graph-data side, then a fresh policy-engine container might just lose the cache from the last healthy graph-data.  Are there any deadlock-y sorts of things we're worried about?